email/src/main/java/edu/jhuapl/dorset/demos/EmailManager.java

```diff
@@ -0,0 +1,393 @@
+
+    private String user;
+    private String password;
+    private String mailStoreType;
+    private String host;
+    
+    private Session session;
+    private Store store;
+    private Folder inboxFolder;
+    private Folder errorFolder;
+    private Folder completeFolder;
+    private Message[] unreadMessages = new Message[1];
+    private Message[] processingMessages = new Message[1];
+    private ArrayDeque<Message> handler = new ArrayDeque<Message>();
+    private static final int HOLDS_MESSAGES = 1;
+    private String response = "";
+    
+    /**
+    * EmailManager Constructor
+    * 
+    * @param config  Configuration object that stores mail server information, username, and password
+    */
+    public EmailManager(Config config) {
+        
+        user = config.getString(USERNAME_KEY);
+        password = config.getString(PASSWORD_KEY);
+        mailStoreType = config.getString(MAIL_STORE_TYPE_KEY);
+        host = config.getString(HOST_KEY); 
+        Properties prop = extractProperties(config);
+        
+        try {
+            session = Session.getDefaultInstance(prop);
+            store = session.getStore(mailStoreType);
+            store.connect(host, user, password);
+        } catch (MessagingException e) {
+            logger.error("Failed to start email session");
+        }
+    }
+    
+    /**
+     * Extract properties from configuration object
+     * 
+     * @param config  Configuration object that stores mail server information, username, and password
+     * @return the properties file
+     */
+    private Properties extractProperties(Config config) {
+        Properties prop = new Properties();
+        for (java.util.Map.Entry<java.lang.String, ConfigValue> entry : config.entrySet()) {
+            prop.setProperty(entry.getKey(), config.getString(entry.getKey()));
+        }        
+        return prop;
+    }
+    /**
+     * Retrieve folders and create them if they do not already exist
+     */
+    public void initFolders() {
+        try {
+            if (!store.getFolder(EmailType.ERROR.getValue()).exists()) {
+                store.getFolder(EmailType.ERROR.getValue()).create(HOLDS_MESSAGES);
+            }
+            if (!store.getFolder(EmailType.COMPLETE.getValue()).exists()) {
+                store.getFolder(EmailType.COMPLETE.getValue()).create(HOLDS_MESSAGES);
+            }
+            errorFolder = store.getFolder(EmailType.ERROR.getValue());
+            completeFolder = store.getFolder(EmailType.COMPLETE.getValue());
+            inboxFolder = store.getFolder(EmailType.INBOX.getValue());
+        } catch (MessagingException e) {
+            logger.error("Failed to initialize folders");
+        }
+    }
+
+    /**
+     * Determine which folder should be accessed
+     * 
+     * @param folder   the folder to be returned. Expressed by an integer value
+     * @return Folder   the folder to be accessed
+     */
+    public Folder determineFolder(EmailType folder) {
```

Comment:
Make this a private method since it is an internal method. Also rename as getFolder() since it returns a Folder. determine is not clear to most developers what it does.